### PR TITLE
ToString() performance improvement and code cleanup

### PR DESCRIPTION
**Pull request by yariks5s**
### Changelog category (leave one):
- Performance Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Since we use separate instructions for executing `toString()` with datetime argument, it is possible to improve performance a bit for non-datetime arguments and have some parts of the code cleaner. Follows up #53680.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
This is an automated comment for commit 66479b69660e999c0f0a58a7aa976e1714869fbb with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/54443/66479b69660e999c0f0a58a7aa976e1714869fbb/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: src/Functions/FunctionsConversion.h [hunk 1]
```diff
@@ -990,20 +990,11 @@ struct ConvertImpl<FromDataType, DataTypeString, Name, ConvertDefaultBehaviorTag
         {
             ColumnUInt8::MutablePtr null_map = copyNullMap(arguments[0].column);
 
-            const auto & col_with_type_and_name =  columnGetNested(arguments[0]);
+            const auto & col_with_type_and_name = columnGetNested(arguments[0]);
             const auto & type = static_cast<const FromDataType &>(*col_with_type_and_name.type);
 
             const DateLUTImpl * time_zone = nullptr;
```
Comment (nickitat) [inline src/Functions/FunctionsConversion.h:996]: not needed anymore?

File: src/Functions/FunctionsConversion.h [hunk 2]
```diff
@@ -990,20 +990,11 @@ struct ConvertImpl<FromDataType, DataTypeString, Name, ConvertDefaultBehaviorTag
...
 
-            if constexpr (std::is_same_v<FromDataType, DataTypeDate> || std::is_same_v<FromDataType, DataTypeDate32>)
```
Comment (nickitat) [inline src/Functions/FunctionsConversion.h:998]: we could add assert like `!IsDataTypeDateOrDateTime<FromDataType>` to make sure this code is never actually needed

Comment (yariks5s) [reply src/Functions/FunctionsConversion.h:998]: [Here](https://github.com/yariks5s/ClickHouse/blob/66479b69660e999c0f0a58a7aa976e1714869fbb/src/Functions/FunctionsConversion.h#L888) we have check regarding `IsDataTypeDateOrDateTime`, so as far as I concerned this check is useless, maybe I am wrong.

File: src/Functions/FunctionsConversion.h (hunk 1 above)
Comment (yariks5s) [reply src/Functions/FunctionsConversion.h:996]: just an extra whitespace.

Comment (nickitat) [reply src/Functions/FunctionsConversion.h:996]: I'm talking about `time_zone` variable

File: src/Functions/FunctionsConversion.h (hunk 2 above)
Comment (nickitat) [reply src/Functions/FunctionsConversion.h:998]: Nope, you're right

File: src/Functions/FunctionsConversion.h (hunk 1 above)
Comment (yariks5s) [reply src/Functions/FunctionsConversion.h]: Just removed it, thanks

**Pull request merged**